Khan/perseus
# Fix bug where arrow keys did not always move segment

## Summary:
On some graphs, after focusing on the line segment, using the arrow keys would not move the segment. After fixing the line component to consider the snap step and adjusting dragging movement to be based on start point instead of midpoint, the line segments now move with all directional keys.

Issue: LEMS-1970

## Test plan:
- On this branch, open localhost with `yarn dev`
- Add in the following content in the flipbook and confirm all segments move in all 4 directions with all 4 arrow keys
```
{"content":"The transformation $\\text E\\,\\vec{x}$ is defined by the following matrix.\n\n$\\text E = \\left[\\begin{array}{rr}0 & 5 \\\\ -5 & 5\\end{array}\\right]$\n\n**Plot the result of $\\text E\\,\\vec{v}$ for $\\vec{v}=\\left[\\begin{array}{r}5 \\\\ 2\\end{array}\\right]$.** \n\n[[☃ interactive-graph 1]]\n\n","images":{},"widgets":{"interactive-graph 1":{"alignment":"default","graded":true,"options":{"backgroundImage":{"height":425,"url":"web+graphie://ka-perseus-graphie.s3.amazonaws.com/e230590dbd3ad30f570b592c71475f899f110097","width":425},"correct":{"coords":[[[0,0],[10,-15]]],"type":"segment"},"graph":{"type":"segment"},"gridStep":[5,5],"labels":["x","y"],"markings":"none","range":[[-25,25],[-25,25]],"rulerLabel":"","rulerTicks":10,"showProtractor":false,"showRuler":false,"showTooltips":false,"snapStep":[5,5],"step":[5,5]},"static":false,"type":"interactive-graph","version":{"major":0,"minor":0}}}}
{"content":"The graph of $y=h(x)$ is a line segment joining the points $(1,9)$ and $(3,2)$. \n\n**Drag the endpoints of the segment below to graph $y=h^{-1}(x)$.\n **\n\n[[☃ interactive-graph 1]]","images":{},"widgets":{"interactive-graph 1":{"alignment":"default","graded":true,"options":{"backgroundImage":{"bottom":0,"height":425,"left":0,"scale":1,"url":"web+graphie://ka-perseus-graphie.s3.amazonaws.com/a82b4098da55c75b68d2c1fd9f348923b8935ccb","width":425},"correct":{"coords":[[[9,1],[2,3]]],"type":"segment"},"graph":{"type":"segment"},"gridStep":[1,1],"labels":["x","y"],"markings":"none","range":[[-100,100],[-100,100]],"rulerLabel":"","rulerTicks":10,"showProtractor":false,"showRuler":false,"snapStep":[10,10],"step":[10,10]},"type":"interactive-graph","version":{"major":0,"minor":0}}}}
{"content":"Line segment $\\overline{OG}$ is rotated $180^\\circ$ about the point $(-2,4)$.  \n\n**Draw the image of this rotation using the interactive graph.**\n\n*The direction of a rotation by a positive angle is counter-clockwise.* \n\n[[☃ interactive-graph 1]]\n\n","images":{},"widgets":{"interactive-graph 1":{"graded":true,"options":{"backgroundImage":{"bottom":0,"height":425,"left":0,"scale":1,"url":"web+graphie://ka-perseus-graphie.s3.amazonaws.com/567155d84afcfa01baf0a44afd994fce3df17b5c","width":425},"correct":{"coords":[[[-7,7],[2,5]]],"type":"segment"},"graph":{"type":"segment"},"gridStep":[1,1],"labels":["x","y"],"markings":"none","range":[[-150,150],[-150,150]],"rulerLabel":"","rulerTicks":10,"showProtractor":false,"showRuler":false,"snapStep":[20,20],"step":[10,10]},"type":"interactive-graph","version":{"major":0,"minor":0}}}}
```
- Confirm all tests pass
- Confirm added tests are good regression tests
- After deploying, confirm graphs 2, 4, 5, and  7 now move on the following page: https://www.khanacademy.org/devadmin/content/articles/segment-graph/xa85475650c4b3749
